Feature/app.

matipojo opened the pull request.

joshmarom committed:
AddNew button component

And reorder package alphabetical order

joshmarom committed:
Fix side menu-item appearance

joshmarom committed:
Change class name of promotion page

matipojo committed:
App/Site Editor: Change - Extract "All Parts" button as a component.

matipojo committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

yonatanelm committed:
# Add Popover pattern (ongoing)
* Add Definition
* Add private API

yonatanelm committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app

yonatanelm committed:
# Add Popover pattern (ongoing)
* update hover state

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

joshmarom committed:
Merge remote-tracking branch 'yonatanelm/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

yonatanelm committed:
# Update Popover pattern (ongoing) : dark theme, arrow
* Add arrow
* Add dark theme

matipojo committed:
App/Site Editor: New - Templates tooltip data.

matipojo committed:
App/Site Editor: Fix - Grunt loader error.

matipojo committed:
App/Site Editor: Fix - Lint errors.

matipojo committed:
App/Loader: Change - Add inline style.

matipojo committed:
App: Restore - Wrong merge.

joshmarom committed:
Add more features to site part component

joshmarom committed:
Fix after merge

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

yonatanelm committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app

yonatanelm committed:
# Remove utils
* remove utils loaders.

joshmarom committed:
Code fixes for site-part component

joshmarom committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
App/Site Editor: Fix - Templates video URL.

matipojo committed:
App: New - Modal Provider component.

matipojo committed:
App/Site Editor: New - Site part info modal.

matipojo committed:
App/Modal: Fix - Use eps z-index.

matipojo committed:
App/Modal: Fix - Add close button.

joshmarom committed:
Improve code for card components

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

yonatanelm committed:
# Grid
* fix broken build due to utils referance in app
* add Grid
* notice: Grid uses some hardcoded values that should be replaced.

yonatanelm committed:
# Grid Classes and vars adjustments
* Fix Custom vars system
* Add mixins for spacing

yonatanelm committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app

joshmarom committed:
Finish refactoring card component

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
App/Loader: Fix - Go Back method.

joshmarom committed:
Added a simple CSS-Grid component

joshmarom committed:
Implement CSS Grid component

joshmarom committed:
Merge remote-tracking branch 'yonatanelm/feature/app' into feature/app

matipojo committed:
App/Site Editor: New - Admin bar menu.

matipojo committed:
App: Fix - Close button on directly mode.

matipojo committed:
App: New - Load via JS API Commands with shortcuts (ctrl+shift+e).

matipojo committed:
App: Fix/Revert - Close button on directly mode.

joshmarom committed:
Make menu action-button dynamic

joshmarom committed:
Styling progress

joshmarom committed:
Screen-only utill

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

matipojo committed:
Templates/Local: New - Add modified date.

matipojo committed:
App/Model: Fix - Modal not closed on overlay click.

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

joshmarom committed:
Coding

joshmarom committed:
Coding

joshmarom committed:
Popover Component

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
App/Site Editor: Fix - Docs URLs.

joshmarom committed:
Fix display issue with button class applied to link tag

joshmarom committed:
Promotion page styling fixes

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

joshmarom committed:
Site-editor menu styling fixes

joshmarom committed:
Site Part progress

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
App/Site Editor: Fix - Admin menu wrong position.

matipojo committed:
App: Fix - Iframe wrong position if page is scrolled.

rotemee committed:
Adding grid component with style props support.

matipojo committed:
Editor: Fix - Site Editor menu item.

rotemee committed:
Changing spacing prop to use css var value and changing nested props to be modifers.

rotemee committed:
Refactoring props classes creation.

rotemee committed:
Implementin import export code in last version of josh feature-app.

rotemee committed:
Implementing grid component in the last version of josh feature-app.

matipojo committed:
Webpack: Fix - Plugins are not available for production mode.

joshmarom committed:
Merge pull request #14 from rotemee/feature/app-grid

Adding grid component with style props support.

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

kobizz commented on core/app/assets/js/layout/header.js:
```diff
@@ -0,0 +1,34 @@
+	} );
+
+	return (
+		<header className="elementor-app__header dialog-header dialog-lightbox-header">
```
We have to decide about all classes prefixes

joshmarom replied:
Any new insights in regards to this ?

joshmarom replied:
done

matipojo commented on assets/dev/scss/global/global.scss:
```diff
@@ -4,6 +4,7 @@
 @import "aspect_ratio";
 @import "flatpicker";
 @import "modal-wrapper";
+@import "core/app/modules/site-editor/assets/scss/admin/admin-bar";
```
Where should it be?

matipojo replied:
@kobizz

rotemee committed:
Working and updated version from josh feature-app.

joshmarom committed:
Make grid gutter rem based

rotemee committed:
Adding dark mode support.

rotemee committed:
Removing ui footer component, changing main footer to footer.

matipojo committed:
App: Change - Temporary loading UI.

matipojo committed:
App: Fix - Document title.

matipojo committed:
App/UI: Fix - Button doesn't open external URLs.

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

yonatanelm committed:
# Typography Atoms
* Add Atomize function
* Add Atom map wit typography key
* Add creat-atom mixin
* update typography
* update custom properties
* update theme-elements-colors
* update dark theme with above mechanism
* Add shame for misc type classes, and port from typography.

yonatanelm committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/import-export

# Conflicts:
#	core/app/assets/js/ui/grid/grid.js
#	core/app/assets/styles/components/_grid.scss

joshmarom committed:
Toggle dark mode based on user preferences

joshmarom committed:
Site editor popover style fine-tuning

joshmarom committed:
Site Editor: Improve card style

joshmarom committed:
Site Editor: Improve card style

joshmarom committed:
Site Editor: coding

joshmarom committed:
Site Editor: Improve `AddNewButton` component

rotemee committed:
Supporting empty features list in the kit content data.

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/import-export

# Conflicts:
#	core/app/assets/js/layout/page.js
#	core/app/assets/styles/app-imports.scss

rotemee committed:
Removing unnecessary prop types checking.

rotemee committed:
Changing data structure.

joshmarom committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

joshmarom committed:
Merge remote-tracking branch 'yonatanelm/feature/app' into feature/app

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/import-export

rotemee committed:
After pull of typography and moving heading and title components to the app level.

joshmarom committed:
Add grid component to the package

joshmarom committed:
Progress with Modal component

kobizz commented on core/app/assets/styles/components/_grid.scss:
```diff
@@ -0,0 +1,226 @@
+
+//@prop: [xs={@num<=12}],sm={@num<=12},md={@num<=12},lg={@num<=12},xl={@num<=12},xxl={@num<=12}]
+//@markup <Grid item xs={4}> or <Grid item sm={3}> and so on..
+@media screen and (min-width: $editor-screen-sm-min) {
+	@include grid-item-auto(sm);
+}
+
+@media screen and (min-width: $editor-screen-md-min) {
+	@include grid-item-auto(md);
+}
+
+@media screen and (min-width: $editor-screen-lg-min) {
+	@include grid-item-auto(lg);
+}
+
+@media screen and (min-width: $editor-screen-xl-min) {
+	@include grid-item-auto(xl);
+}
+
+@media screen and (min-width: $editor-screen-xxl-min) {
+	@include grid-item-auto(xxl);
+}
```
Do it with a loop

kobizz replied:
Done

kobizz commented on core/app/assets/styles/components/_grid.scss:
```diff
@@ -0,0 +1,226 @@
+// Generating cols per each media breakpoint
+@include grid-col-generator(xs);
+
+@media screen and (min-width: $editor-screen-sm-min) {
+	@include grid-col-generator(sm);
+}
+
+@media screen and (min-width: $editor-screen-md-min) {
+	@include grid-col-generator(md);
+}
+
+@media screen and (min-width: $editor-screen-lg-min) {
+	@include grid-col-generator(lg);
+}
+
+@media screen and (min-width: $editor-screen-xl-min) {
+	@include grid-col-generator(xl);
+}
+
+@media screen and (min-width: $editor-screen-xxl-min) {
+	@include grid-col-generator(xxl);
+}
```
Loop

kobizz replied:
Done

kobizz commented on core/app/assets/styles/functions/_size.scss:
```diff
@@ -0,0 +1,10 @@
+
+@function size($multiplier, $unit:0.5rem) {
+    @return $multiplier * $unit;
+}
```
Add an empty line at end of the file

kobizz commented on core/app/assets/styles/mixins/_clearfix.scss:
```diff
@@ -0,0 +1,8 @@
+// Mixins: Clearfix
+@mixin clearfix() {
```
Why do we need this?

kobizz replied:
Remove it

joshmarom replied:
done

kobizz commented on core/app/assets/styles/mixins/_screen-reader.scss:
```diff
@@ -0,0 +1,34 @@
+// See: https://a11yproject.com/posts/how-to-hide-content/
+// See: https://hugogiraudel.com/2016/10/13/css-hide-and-seek/
+
+@mixin sr-only() {
```
There is already "sr" handling in this pr

kobizz replied:
Remove duplication

joshmarom replied:
done

kobizz commented on core/app/assets/styles/mixins/_text-hide.scss:
```diff
@@ -0,0 +1,8 @@
+// Mixins: Text-gide , for image replacement
+@mixin text-hide() {
```
What is it used for?

kobizz replied:
@yonatanelm

kobizz replied:
Remove it

kobizz commented on core/app/assets/styles/mixins/_text-hide.scss:
```diff
@@ -0,0 +1,8 @@
+// Mixins: Text-gide , for image replacement
```
```suggestion
// Mixins: Text-hide , for image replacement
```

joshmarom replied:
done

kobizz commented on core/app/assets/styles/mixins/_text-truncate.scss:
```diff
@@ -0,0 +1,8 @@
+// Utilities: Text truncate
+// Requires inline-block or block for proper styling
+
+@mixin text-truncate() {
```
?

joshmarom replied:
see site-template metadata

joshmarom replied:
done

kobizz commented on core/app/assets/styles/tokens/_ratio.scss:
```diff
@@ -0,0 +1,15 @@
+$embed-responsive-aspect-ratios: () !default;
+// stylelint-disable-next-line scss/dollar-variable-default
+$embed-responsive-aspect-ratios: join(
+						(
+										(21 9),
+										(16 9),
+										(4 3),
+										(1 1),
+						),
+				$embed-responsive-aspect-ratios
```
Fix the indentation

joshmarom replied:
done

kobizz commented on core/app/assets/styles/tokens/_spacing.scss:
```diff
@@ -0,0 +1 @@
+$eps-spacer: 0.5rem;
```
What is it used for?

joshmarom replied:
To make a good mathematical base spacing unit for multiplication

kobizz commented on core/app/assets/styles/tokens/_utilities.scss:
```diff
@@ -0,0 +1,19 @@
+$sizes: () !default;
+// stylelint-disable-next-line scss/dollar-variable-default
+$sizes: map-merge(
+				(
+						25: 25%,
+						50: 50%,
+						75: 75%,
+						100: 100%,
+						auto: auto
+				),
+				$sizes
```
Indentation

joshmarom replied:
done

joshmarom committed:
Feature/App: Fix Grid components

joshmarom committed:
Feature/App: Modal Progress

joshmarom committed:
Feature/App: Improve grid coding

joshmarom committed:
Feature/App: Modal Progress

rotemee committed:
Adding typography handling.

rotemee committed:
Removing import-export module to only get the app level components.

rotemee committed:
Removing import-export files calls.

rotemee committed:
Removing import export css files calls.

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app-features

rotemee committed:
Fixing props types.

rotemee committed:
Test

rotemee committed:
Test

rotemee committed:
Test

rotemee committed:
Props test

joshmarom committed:
Merge pull request #16 from rotemee/feature/app-features

Feature/app features

joshmarom committed:
Feature/App: Include New Typography components in app package

joshmarom committed:
Feature/App: Typography

joshmarom committed:
Feature/App: Buttons Improvements

joshmarom committed:
Feature/App: Site Editor Progress

joshmarom committed:
Button link style

kobizz commented on core/app/view.php:
```diff
@@ -0,0 +1,26 @@
+	<?php wp_print_styles(); ?>
+</head>
+<body>
+<div id="elementor-app"></div>
```
e-app

joshmarom replied:
done

kobizz commented on core/editor/editor.php:
```diff
@@ -496,6 +496,7 @@ public function enqueue_scripts() {
 				'jquery-hover-intent',
 				'nouislider',
 				'pickr',
+				'react',
```
?

matipojo replied:
Until now it was included via webpack, but for caching for both editor & app frames - it's better to include it as an external file.

kobizz reviewed (changes requested):
You should remove all empty files

joshmarom committed:
Feature/App: Design system fixes

joshmarom commented on core/app/app.php:
```diff
@@ -0,0 +1,179 @@
+
+	protected function get_init_settings() {
+		return [
+			'menu_url'  => $this->get_base_url() . '#site-editor/promotion',
+			'assets_url'  => ELEMENTOR_ASSETS_URL,
+			'return_url'  => isset( $_SERVER['HTTP_REFERER'] ) ? $_SERVER['HTTP_REFERER'] : admin_url(),
```
@matipojo

kobizz replied:
@matipojo ?

matipojo replied:
It's complicated, 3rd plugins may change the URLs via `set_settings` that doesn't support nested valuse.

kobizz replied:
You can always change the value by filter

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,99 @@
+$app-sidebar-dark-background-color: dark-tints(700);
+$box-shadow-dark-color: rgba(theme-colors(dark), $opacity-015);
+
+.dark {
```
I did it already

joshmarom replied:
done

kobizz replied:
Change it to `.eps-theme-dark`

joshmarom replied:
done

joshmarom commented on core/app/assets/js/ui/molecules/add-new-button.js:
```diff
@@ -0,0 +1,33 @@
+
+import './add-new-button.scss';
+
+export default class AddNewButton extends Button {
```
Good question

joshmarom replied:
@matipojo I think it should be a reusable component.
Do you have any strong reason why not?

matipojo replied:
It can be a reusable component, but from the `site-editor` package. it's not related to our general components.

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

joshmarom committed:
Feature/App: Refactor some typography-related token names

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

joshmarom committed:
Feature/App: Improvements, refactoring and cleanup

joshmarom committed:
Resolve Conflict

joshmarom committed:
Feature/App: Dark Mode Progress

joshmarom committed:
Feature/App: Modal private API and dark theme

joshmarom committed:
File structure and organize

rotemee committed:
Adding Grid to css imports.

joshmarom committed:
Feature/App: Progress toward completion

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
App/Site Editor: Fix - JSCS.

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app-features

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

matipojo committed:
Merge branch 'hotfix/move-getUniqueID-to-elementor-common' into feature/app

joshmarom committed:
Cleanup and encapsulation of app style

joshmarom committed:
Typography settings maintenance

joshmarom committed:
Remove redundant import call

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

joshmarom committed:
Feature/App: Remove prop `ghost` prop from button component

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

joshmarom committed:
Faeture/App: Cleanup and styling finishes

joshmarom committed:
Feature/App: Crazy genius CSS Grid stuff

joshmarom committed:
Feature/App: Cleanup and finishes

joshmarom committed:
Feature/App: Polish looks of the app for delivery

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app-features

# Conflicts:
#	core/app/assets/styles/app-imports.scss

yonatanelm committed:
# App Scss partial: private Api and cleanups
* Move not related definitions to shame.scss
* Create and link private API partial.
* Map global tokens to private keys
* Update dark, default themes custom properties
* Fix shadows: add global custom properties and dark tokens.
* Add missing custom properties to global elements map
* Move HR  tag to reset
* Fix font definitions to include family sets.

joshmarom committed:
Merge pull request #17 from yonatanelm/feature/app

# App Scss partial: private Api and cleanups

yonatanelm committed:
# App Scss partial: custom property typo
* Fix: global shadow and 'hr' custom properties names

yonatanelm committed:
# App Scss partial: global dark theme issues
* Fix dark theme map, revert invert for dark/light

joshmarom committed:
Merge pull request #18 from yonatanelm/feature/app

# App Scss partial: custom property typo

joshmarom committed:
Merge remote-tracking branch 'yonatanelm/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

joshmarom committed:
Feature/App: Box shadow fixes

rotemee committed:
Merge branch 'feature/app' of https://github.com/joshmarom/elementor into feature/app-features

rotemee committed:
Adding a footer component in the app level.

rotemee committed:
Changing app main element overflow to overlow y.

rotemee committed:
Adding border-radius also to contained button.

rotemee committed:
Updating line height value of heading 3.

joshmarom committed:
Feature/App: Try to match url in menu for active flag

joshmarom committed:
Merge remote-tracking branch 'rotemee/feature/app-features' into feature/app

matipojo committed:
Merge branch 'release/3.0.0' into feature/app

joshmarom committed:
Feature/App: Make menu item active state work

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

joshmarom committed:
Refactor classNames for menu components

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

joshmarom committed:
WIP Menu active flag

joshmarom committed:
Temporary fix for active-menu-item issues

joshmarom committed:
Merge branch 'feature/eps-demo' into feature/app

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

matipojo committed:
App: Fix - Don't load react via webpack.

matipojo committed:
App: Fix - Don't use `StrictMode` for production.

matipojo committed:
App: Fix - HTML CS.

joshmarom commented on core/app/assets/js/layout/header-buttons.js:
```diff
@@ -0,0 +1,44 @@
+	return (
+		<div className="elementor-templates-modal__header__items-area">
+			<Button
+				id="close"
```
Check if we even need an id for this button

joshmarom replied:
done

matipojo commented on core/app/assets/js/layout/header-button.js:
```diff
@@ -0,0 +1,15 @@
+	} );
+
+	getCssId() {
+		return `eps-app-headerBtn-` + super.getCssId();
```
```suggestion
		return `eps-app-header-btn-` + super.getCssId();
```

joshmarom replied:
Done.
Note we need to decide what to do about class components ... we have a few of them

matipojo commented on core/app/assets/js/layout/page.js:
```diff
@@ -0,0 +1,58 @@
+import Footer from './footer';
+
+export default function Page( props ) {
+	const getSideBar = () => (
```
```suggestion
	const Sidebar = () => (
```

joshmarom replied:
It's a problem because we import sidebar and footer components in this component, so the names clash.
I used `AppSidebar` and `AppFooter` as var names. 
Let me know if you have a different idea

matipojo commented on core/app/assets/js/layout/page.js:
```diff
@@ -0,0 +1,58 @@
+			{ props.sidebar }
+		</Sidebar>
+	),
+	getFooter = () => (
```
```suggestion
	Footer = () => (
```

joshmarom replied:
See above

matipojo commented on core/app/assets/js/layout/page.js:
```diff
@@ -0,0 +1,58 @@
+				<div className="eps-app">
+					<Header title={ props.title } buttons={ props.headerButtons } />
+					<div className="eps-app__main">
+						{ props.sidebar && getSideBar() }
+						<Content>
+							{ props.content }
+						</Content>
+					</div>
+					{ props.footer && getFooter() }
```
Check the prop from within the function,

joshmarom replied:
done

matipojo commented on core/app/assets/js/loader/commands/load.js:
```diff
@@ -0,0 +1,33 @@
+		if ( ! component.iframe ) {
+			component.iframe = document.createElement( 'iframe' );
+			component.iframe.className = 'elementor-app-iframe';
+			component.iframe.style.cssText = '' +
+				'display: none;' +
+				'width: 100%;' +
+				'height: 100%;' +
+				'position: fixed;' +
+				'top: 0;' +
+				'left: 0;' +
+				'z-index: 99999; /* Over WP Admin Bar */' +
+				'background-color: rgba(0, 0, 0, 0.8);';
```
Why load a CSS file for one element?

matipojo commented on core/app/assets/js/pages/not-found.scss:
```diff
@@ -0,0 +1,7 @@
+.eps-app__not-found .eps-app__sidebar {
+	display: none;
```
The element shouldn't be exist.

joshmarom replied:
done

matipojo commented on core/app/assets/js/pages/not-found.scss:
```diff
@@ -0,0 +1,7 @@
+}
+
+.eps-app__not-found .eps-app__content {
+	width: 100%;
```
Can it be handled with flex grows?

joshmarom replied:
done

rotemee committed:
Fixing app main and content scroll bars.

rotemee committed:
Changing Text component children type to any to handle multiple variations at the same time.

rotemee committed:
Updating heading display-3 color.

rotemee committed:
Uncommenting the Button component dark mode of custom properties.

rotemee committed:
Implementing select and select2 components.

matipojo committed:
Document: Fix - Remove unused `get_icon` method.

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

rotemee committed:
Changing select2 to export default and adding support with the multiple attribute.

matipojo committed:
Webpack: Fix - ReactDOM should be loaded as the React library.

Load them externally in order to share the assets between the editor and the app.

matipojo committed:
Package.json: Fix - Cleanup unused dependencies.

rotemee committed:
Removing value prop default value: can't be string when using multiple.

Nevoss committed:
App : Data - Fix - base endpoint registered without any event + add 'wp-api-request' to elementor-common.

Nevoss committed:
Merge branch 'feature/app' of github.com:elementor/elementor into feature/app

matipojo committed:
Tests: Fix - ReactDOM extenal file is missing.

matipojo committed:
Merge remote-tracking branch 'Nevoss/feature/app' into feature/app

matipojo committed:
Webpack: Fix - ReactDOM is missing on build.

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/eps-demo

joshmarom committed:
Merge remote-tracking branch 'rotemee/feature/app-features' into feature/eps-demo

joshmarom committed:
Code cleanup

joshmarom committed:
Merge branch 'feature/eps-demo' into feature/app

joshmarom committed:
Site editor touchups and cleanups

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

rotemee committed:
Removing font-size prop from .eps-app.

joshmarom committed:
App style

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

rotemee committed:
Adding underlined variant to button.

rotemee committed:
Creating Utils class and replacing CSS Grid inner function with the utility function.

rotemee committed:
Changing text entities colors in dark mode.

joshmarom committed:
App: Fix issue with action button color

joshmarom committed:
Site Editor: Rename dialog input (temporary solution)

joshmarom committed:
Merge remote-tracking branch 'rotemee/feature/app-features' into feature/app

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,124 @@
+@import "../styles/shame";
+
+
+.eps-app {
```
It's a temporary file for some css TBD what to do about this file's contents...

joshmarom commented on core/app/assets/js/ui/atoms/text.js:
```diff
@@ -0,0 +1,29 @@
+	classes.push( baseClassName + '-text' + variant );
+
+	return React.createElement( tagName, {
+		className: classes.filter( ( classItem ) => '' !== classItem ).join( ' ' ),
```
done

joshmarom commented on core/app/assets/js/ui/atoms/icon.js:
```diff
@@ -0,0 +1,15 @@
+import './icon.scss';
```
done

joshmarom commented on core/app/assets/js/ui/atoms/heading.js:
```diff
@@ -0,0 +1,31 @@
+	}
+
+	return React.createElement( tagName, {
+		className: classes.filter( ( classItem ) => '' !== classItem ).join( ' ' ),
```
done

joshmarom commented on core/app/assets/js/ui/atoms/heading.js:
```diff
@@ -0,0 +1,31 @@
+}
+
+Heading.propTypes = {
+	className: PropTypes.string,
```
done

joshmarom commented on core/app/assets/js/ui/atoms/text.js:
```diff
@@ -0,0 +1,29 @@
+		classes = [
+			props.className,
+		],
+		tagName = props.tag || 'p',
```
Sorry, not sure what you mean

matipojo replied:
You can set the default  `props.tag` to `'p'`.

joshmarom replied:
done

joshmarom commented on core/app/assets/js/ui/atoms/typography.js:
```diff
@@ -0,0 +1,21 @@
+import './typography.scss';
+
+export default function Typography( props ) {
```
An Early attempt at doing the `<Text>` component.
I will remove it

kobizz replied:
Remove it

joshmarom replied:
done

joshmarom commented on core/app/assets/js/ui/atoms/typography.js:
```diff
@@ -0,0 +1,21 @@
+import './typography.scss';
```
done

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,99 @@
+	display: flex;
+	align-items: center;
+	justify-content: center;
+}
```
done

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,99 @@
+h1 {
+	font-size: var(--h1-font-size);
+	margin-bottom: var(--h1-margin-bottom);
+}
```
done

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,99 @@
+hr {
+	border: 0 none;
+	border-bottom: 1px solid var(--hr-color);
+}
```
moved to `shame.scss` TBD

joshmarom replied:
done

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,99 @@
+		height: 100%;
+		overflow: scroll;
+	}
+}
```
We discussed this  and made a separation between *Structure* configuration and *Stylistic* ones.
We should write the guidelines with examples

joshmarom replied:
done

joshmarom commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,99 @@
+	}
+}
+
+.dialog-type-lightbox:not(.elementor-popup-modal) {
```
done

joshmarom commented on core/app/assets/js/layout/header-button.js:
```diff
@@ -0,0 +1,15 @@
+	} );
+
+	getCssId() {
+		return `elementor-template-library-header-` + super.getCssId();
```
done

joshmarom commented on core/app/assets/js/layout/header-button.js:
```diff
@@ -0,0 +1,15 @@
+	}
+
+	getClassName() {
+		return `elementor-templates-modal__header__item ` + super.getClassName();
```
done

joshmarom commented on core/app/assets/js/layout/header-buttons.js:
```diff
@@ -0,0 +1,44 @@
+import Button from './header-button';
+
+export default function HeaderButtons( props ) {
```
done

joshmarom commented on core/app/assets/js/ui/card/card-overlay.js:
```diff
@@ -0,0 +1,18 @@
+
+CardOverlay.propTypes = {
+	className: PropTypes.string,
+	children: PropTypes.object.isRequired,
```
done

joshmarom commented on core/app/assets/js/ui/grid/grid.scss:
```diff
@@ -0,0 +1,198 @@
+			flex-wrap: wrap-reverse;
+		}
+
+		//@prop: spacing[5,8,10,12,16]
```
done

joshmarom commented on core/app/assets/js/ui/grid/grid.scss:
```diff
@@ -0,0 +1,198 @@
+		flex-wrap: wrap;
+		width: 100%;
+
+		//@prop: no-wrap
```
done

joshmarom commented on core/app/assets/js/ui/grid/grid.scss:
```diff
@@ -0,0 +1,198 @@
+
+// Zero min width to disable overflow while container is no wrap
+
+//@prop: ZeroMinWidth
+//@markup <Grid item zeroMinWidth> or <Grid item sm> and so on..
+.#{$eps-prefix}grid-item--zero-min-width {
+	min-width: 0;
+}
+
+// Item flex sizing, auto grid
+//@prop: [xs,sm,md,lg,xl,xxl]
+//@markup <Grid item xs> or <Grid item sm> and so on..
+// No need for container:
+//@prop: [xs={@num<=12}],sm={@num<=12},md={@num<=12},lg={@num<=12},xl={@num<=12},xxl={@num<=12}]
+//@markup <Grid item xs={4}> or <Grid item sm={3}> and so on..
+@each $breakpoint in map-keys($breakpoints) {
+	@media screen and (min-width: #{map-get($breakpoints, $breakpoint)}) {
+		@include grid-item-auto($breakpoint);
+	}
+}
```
Done

joshmarom commented on core/app/assets/js/ui/molecules/popover.js:
```diff
@@ -0,0 +1,18 @@
+}
+
+Popover.propTypes = {
+	children: PropTypes.any,
```
Done

joshmarom committed:
App: Remove redundant files

joshmarom committed:
App: .eps prefix everywhere

joshmarom committed:
App: code review and cleanup

joshmarom commented on core/app/modules/site-editor/assets/js/organisms/menu.js:
```diff
@@ -0,0 +1,43 @@
+
+	return (
+		<UiMenu menuItems={ templateTypes } actionButton={ actionButton } promotion={ props.promotion }>
+			<>
```
done

joshmarom committed:
App: Hide the shame

joshmarom commented on core/app/assets/js/pages/not-found.scss:
```diff
@@ -0,0 +1,7 @@
+
+.elementor-app__not-found .elementor-app__content {
+	width: 100%;
+}
```
done

joshmarom commented on core/app/assets/js/ui/atoms/css-grid.scss:
```diff
@@ -0,0 +1,9 @@
+$eps-1px: spacing(10) / 10;
```
done

joshmarom commented on core/app/assets/js/ui/atoms/css-grid.scss:
```diff
@@ -0,0 +1,9 @@
+$eps-grid-gap: calc(var(--eps-grid-spacing) * #{$eps-1px});
+$eps-grid-columns: repeat(var(--eps-grid-columns), 1fr);
+
+.eps-css-grid {
```
yes

kobizz replied:
Change this component name to 'grid'.

Change the other component name to 'flex'.

joshmarom replied:
Can we wait a bit longer with that? I don't want to create too much confusion at this moment .

kobizz replied:
We'll postpone it to the next release

joshmarom commented on core/app/assets/js/ui/card/card.scss:
```diff
@@ -0,0 +1,88 @@
+	&__headline {
+		color: var(--card-headline-color);
+		--h1-font-size: $card-headline-font-size;
+		--h1-margin-bottom: 0;
```
done

joshmarom commented on core/app/assets/js/ui/card/card.scss:
```diff
@@ -0,0 +1,88 @@
+	}
+
+	&__info-icon {
+		//color: $card-info-icon-color;
```
done

joshmarom commented on core/app/assets/js/ui/menu/menu-item.js:
```diff
@@ -0,0 +1,12 @@
+import './menu-item.scss';
+import BaseButton from '../molecules/button';
+
+export default class SideMenuItem extends BaseButton {
```
I think so

joshmarom replied:
done

joshmarom commented on core/app/assets/js/ui/menu/menu.scss:
```diff
@@ -0,0 +1,44 @@
+@import "../../../styles/common";
+
+$menu-title-font-size: 11px;
+$menu-title-line-height: 13px;
```
done

joshmarom commented on core/app/assets/styles/_components.scss:
```diff
@@ -0,0 +1 @@
+
```
done

joshmarom commented on core/app/assets/styles/tokens/maps/_colors-map.scss:
```diff
@@ -0,0 +1,16 @@
+// Maps
```
@yonatanelm

joshmarom commented on core/app/assets/styles/tokens/maps/_z-index-map.scss:
```diff
@@ -0,0 +1,10 @@
+// Maps: z-index map
```
@yonatanelm

joshmarom commented on core/app/assets/styles/tokens/styles/_radius.scss:
```diff
@@ -0,0 +1,7 @@
+// EDS radius tokens
+$eps-radius: .1875rem;//~3px
```
@yonatanelm

joshmarom commented on core/app/assets/styles/tokens/styles/_shadow.scss:
```diff
@@ -0,0 +1,19 @@
+
+$eps-box-shadow-sm: null;
+$eps-box-shadow-lg: null;
+$eps-box-shadow-hover: null;
```
@yonatanelm

kobizz replied:
Remove `null` tokens

joshmarom commented on core/app/modules/site-editor/assets/js/molecules/site-part.scss:
```diff
@@ -0,0 +1,62 @@
+// Site Part
+
+// this should emulate external packages consumption
+@import "../../../../../assets/styles/common";
```
done

joshmarom commented on core/app/modules/site-editor/assets/js/organisms/site-parts.scss:
```diff
@@ -0,0 +1,7 @@
+@import "../../../../../assets/styles/common";
```
done

joshmarom commented on core/app/modules/site-editor/assets/js/pages/promotion.scss:
```diff
@@ -0,0 +1,55 @@
+// Site Part
+
+// this should emulate external packages consumption
+@import "../../../../../assets/styles/common";
```
done

joshmarom replied:
done

joshmarom commented on core/app/modules/site-editor/assets/js/pages/promotion.scss:
```diff
@@ -0,0 +1,55 @@
+	hr {
+		margin-bottom: spacing(44);
+	}
+}
```
done

kobizz replied:
Add `e-` prefix for all none eps classes

joshmarom replied:
done

joshmarom committed:
App: WIP moved dark-mode detection to php level

joshmarom committed:
App: Code Review

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

kobizz commented on core/app/app.php:
```diff
@@ -0,0 +1,179 @@
+			Plugin::$instance->common->register_scripts();
+		}
+
+		$direction_suffix = is_rtl() ? '-rtl' : '';
```
@matipojo

matipojo replied:
Handled https://github.com/elementor/elementor/pull/11806/commits/1c8add023448accf1f4fc112f585556cbd9843b7

matipojo committed:
App: Fix - RTL file is already handled by module.

kobizz commented on core/app/assets/js/ui/card/card-api.scss:
```diff
@@ -0,0 +1,60 @@
+
+$card-header-padding: $card-padding;
+$card-header-border: $eps-border;
+$card-header-height: spacing(20) * 2;
```
@yonatanelm

kobizz replied:
Use a private token interpolation that multiple card padding 4 times

joshmarom replied:
done

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

kobizz commented on core/app/assets/js/ui/modal/modal.scss:
```diff
@@ -0,0 +1,18 @@
+		height: 50%;
+		background: white;
+	}
+}
```
Fix animation API

joshmarom replied:
done

kobizz commented on core/app/assets/js/app.scss:
```diff
@@ -0,0 +1,126 @@
+	width: 100%;
+	max-width: $app-max-width;
+
+	/* Used for boxed version
+
+	@media (min-width: #{map-get($breakpoints, xl)}) {
+		max-width: $app-max-width-min-xl;
+	}
+	*/
```
Remove commented code

kobizz replied:
@shilo-ey What about the full-width mode?

kobizz commented on core/app/assets/js/pages/not-found.js:
```diff
@@ -0,0 +1,14 @@
+import Page from 'elementor-app/layout/page';
```
Do you use it?

kobizz commented on core/app/assets/js/ui/atoms/checkbox-api.scss:
```diff
@@ -0,0 +1,28 @@
+$eps-checkbox-indicator-scale-unit: 1/4;
+$eps-checkbox-indicator-size-unit: ceil($eps-checkbox-indicator-scale-unit * $eps-checkbox-size) - $eps-checkbox-indicator-border-width;
+
+:root {
+  --checkbox-border-color: #{$eps-checkbox-border-color};
+  --checkbox-hover-border-color: #{$eps-checkbox-hover-border-color};
+  --checkbox-active-border-color: #{$eps-checkbox-active-border-color};
+  --checkbox-background-color: #{$eps-checkbox-background-color};
+  --checkbox-checked-background-color: #{$eps-checkbox-checked-background-color};
+  --checkbox-checked-hover-background-color: #{$eps-checkbox-checked-hover-background-color};
+  --checkbox-checked-active-background-color: #{$eps-checkbox-checked-active-background-color};
+  --checkbox-indicator-color: #{$eps-checkbox-indicator-color};
+}
```
Why do you use custom properties?

kobizz commented on core/app/assets/js/ui/card/card.js:
```diff
@@ -0,0 +1,19 @@
+
+export default function Card( props ) {
+	return (
+		<article id={ `type-${ props.type }` } className={ `eps-card ${ props.className }` }>
```
Add a prefix to the id

joshmarom replied:
But the context is not derermeind at this point it's a generic element...

kobizz commented on core/app/assets/js/ui/molecules/add-new-button.scss:
```diff
@@ -0,0 +1,22 @@
+$eps-add-new-button-icon-color: theme-colors(light);
+$eps-add-new-button-size: spacing(24);
+$eps-add-new-button-size-sm: spacing(16);
+$eps-add-new-button-icon-size: calc( var(--eps-add-new-button-size) * 0.75 );
```
Size ratio should be kept in its own variable

joshmarom replied:
done

kobizz commented on core/app/assets/styles/app-imports.scss:
```diff
@@ -0,0 +1,17 @@
+@import "components/_popover.scss";
+@import "../../modules/site-editor/assets/js/organisms/menu.scss";
+@import "../../modules/site-editor/assets/js/pages/promotion.scss";
+@import "../../modules/site-editor/assets/js/context/template-types.scss";
```
@matipojo ?

matipojo replied:
It's an auto-generated file no one should manage it manually, also we don't need two auto-generated files.

kobizz commented on core/app/assets/styles/base/_typography.scss:
```diff
@@ -0,0 +1,122 @@
+.h6 {
+  font-size: $eps-h-5-font-size;
+}
+
```
@yonatanelm

joshmarom replied:
done

kobizz replied:
For now remove the `h` tags, since it causes a CSS duplication

kobizz commented on core/app/assets/styles/base/_typography.scss:
```diff
@@ -0,0 +1,122 @@
+  font-weight: 500;
+  font-size: 14px;
+  line-height: 14px;
+}
```
Done

kobizz commented on core/app/assets/styles/tokens/_z-index.scss:
```diff
@@ -0,0 +1,8 @@
+// Z-INDEX
+$eps-zindex-dropdown: 1000;
```
@yonatanelm

kobizz replied:
Rearrange it

kobizz commented on core/app/modules/site-editor/assets/js/context/template-types.scss:
```diff
@@ -0,0 +1,10 @@
+.e-loading-wrapper {
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	height: 100vh;
+	color: #efebeb;
+	font-family: sans-serif;
+	font-size: 16px;
+	text-shadow: 2px 2px #000;
+}
```
Check if we can remove it

joshmarom replied:
done

matipojo committed:
App: Fix - JS API is not availbe for 3rd party plugins on iframe mode.

Remove the `iframe` mode that cause the `$e` to be available only after the core is loaded.

joshmarom committed:
App: remove redundant file

joshmarom committed:
App: update selector for `eps-theme-dark`

joshmarom committed:
App: remove redundant import of non-existing file

matipojo committed:
App: Fix - Close button not working.

joshmarom committed:
Coding

joshmarom committed:
App: dark-theme flag mechanism refined

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

matipojo committed:
App/Site Editor: Fix - Product icon is missing.

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

joshmarom commented on core/app/assets/styles/base/_typography.scss:
```diff
@@ -0,0 +1,122 @@
+}
+
+p {
+  margin-bottom: spacing(16);
```
done

joshmarom commented on core/app/assets/styles/base/_typography.scss:
```diff
@@ -0,0 +1,122 @@
+
+.text-xl {
+  font-size: $eps-text-xl-font-size;
+}
```
done

matipojo committed:
App/Site Editor: Fix - Error 500 on import template.

joshmarom committed:
App: Code review

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

kobizz commented on core/app/assets/js/ui/modal/modal.scss:
```diff
@@ -0,0 +1,102 @@
+	}
+}
+
+@keyframes pop {
```
Consider adding a prefix to the animation name

joshmarom replied:
Did it and moved it to a global animations file

kobizz commented on core/app/assets/js/ui/molecules/buttons-api.scss:
```diff
@@ -0,0 +1,124 @@
+$button-dark-disabled-active-background-color: darken($button-dark-disabled-background-color, $button-active-constant);
+$button-dark-disabled-color: dark-theme-colors(light);
+
+// using mixing to transfer custom properties
```
Why do you need mixins here?

joshmarom replied:
@yonatanelm did most of the work here...
It needs more work done on it.

kobizz commented on core/app/assets/js/utils/utils.js:
```diff
@@ -0,0 +1,9 @@
+export default class Utils {
+	static pxToRem = ( pixels ) => {
```
Where is this class used?

joshmarom replied:
For spacing in some components.
It needs to be discussed

kobizz commented on core/app/view.php:
```diff
@@ -0,0 +1,38 @@
+if ( 'auto' === $ui_theme ) {
+	wp_register_script( 'detect-dark', '' );
+	wp_enqueue_script( 'detect-dark' );
+	wp_add_inline_script( 'detect-dark', 'if ( window.matchMedia && window.matchMedia(`(prefers-color-scheme: dark)`).matches )
```
Do it where the scripts are enqueued

kobizz commented on core/app/modules/site-editor/assets/js/molecules/site-part.scss:
```diff
@@ -0,0 +1,62 @@
+$indicator-bullet-border: 2px solid var(--indicator-bullet-border-color);
+$indicator-bullet-spacing: spacing(10);
+
+.indicator-bullet {
```
Prefix

kobizz commented on core/app/modules/site-editor/assets/js/molecules/site-part.scss:
```diff
@@ -0,0 +1,62 @@
+	--info-toggle-hover-color: #{$info-toggle-dark-hover-color};
+}
+
+.info-toggle {
```
Prefix

kobizz commented on core/app/modules/site-editor/assets/js/organisms/all-parts-button.js:
```diff
@@ -0,0 +1,29 @@
+
+				return (
+					<MenuItem
+						id="all-parts"
```
Change the ID

kobizz commented on core/app/modules/site-editor/assets/js/pages/promotion.scss:
```diff
@@ -0,0 +1,21 @@
+$promotion-overlay-icon-spacing: spacing(16);
+$promotion-overlay-icon-color: theme-colors(light);
+
+.promotion-overlay {
```
Prefix

kobizz commented on core/app/modules/site-editor/assets/js/templates/site-editor.scss:
```diff
@@ -0,0 +1,6 @@
+.site-editor {
```
Prefix

KingYes committed:
Merge pull request #11983 from matipojo/feature/app

App: Fixes

joshmarom committed:
Improves code review

joshmarom committed:
App: Kobi Code Review

kobizz committed:
Merge branch 'feature/app' of github.com:elementor/elementor into feature/app

* 'feature/app' of github.com:elementor/elementor:
  App/Site Editor: Fix - Error 500 on import template.
  App/Site Editor: Fix - Product icon is missing.
  App: Fix - Close button not working.

joshmarom committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

joshmarom committed:
App: Improve dark theme detection code

kobizz committed:
Merge branch 'feature/app' of github.com:joshmarom/elementor into feature/app

* 'feature/app' of github.com:joshmarom/elementor:
  App: Improve dark theme detection code

joshmarom committed:
Coding

matipojo committed:
Merge remote-tracking branch 'joshmarom/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'elementor/feature/app' into feature/app

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.0.0' into feature/app

matipojo committed:
Site Editor: Change - Rename to Theme Builder.

kobizz committed:
Fixed card typography

KingYes committed:
Merge pull request #12020 from matipojo/hotfix/app

Site Editor: Change - Rename to Theme Builder.

KingYes committed:
Merge pull request #12031 from kobizz/fix/card-typography

Fixed card typography
